lightstep/tracer.py

```diff
@@ -73,8 +73,8 @@ class _LightstepTracer(BasicTracer):
     def __init__(self, enable_binary_format, recorder, scope_manager):
         """Initialize the LightStep Tracer, deferring to BasicTracer."""
         super(_LightstepTracer, self).__init__(recorder, scope_manager=scope_manager)
-        self.register_propagator(Format.TEXT_MAP, TextPropagator())
-        self.register_propagator(Format.HTTP_HEADERS, TextPropagator())
+        self.register_propagator(Format.TEXT_MAP, B3Propagator())
+        self.register_propagator(Format.HTTP_HEADERS, B3Propagator())
```

Comment:
Think we wanted the default here to continue being the TextPropagator

Reply:
This was requested by @carlosalberto before:

> Hey @ocelotl
> 
> Thanks. Actually we need a last iteration, which I hadn't realize. The way injection/extraction works is to specify one of the formats in `opentracing.Format`, and not 'create' a new one to use B3 (or later `TraceContext`), **but** use B3 when either `TEXT_MAP` or `HTTP_HEADERS` are specified:
> 
> ```python
> # replaces the default propagator with the B3 one
> tracer.register_propagator(b3_propagator)
> ...
> # we use the text map format, with the b3 propagator
> tracer.inject(span.context, opentracing.Format.TEXT_MAP, carrier)
> ```
> 
> For this you will need to also update the example ;) Once this is done, we will be able to merge.

Reply:
Ah, my understanding is that both TEXT_MAP and HTTP_HEADERS should default to TextPropagator, unless configured otherwise when configuring the Tracer.

Reply:
@carlosalberto can you confirm, please?

Reply:
Yes, we want `TextPropagator` being the default.

> not 'create' a new one to use B3 (or later TraceContext), but use B3 when either TEXT_MAP or HTTP_HEADERS are specified

Correct, we want `B3` to **completely** replace `TextPropagator` **when** registered explicitly (instead of having it as a separate format).

Reply:
I [reverted](https://github.com/lightstep/lightstep-tracer-python/blob/70da71a1eb75b32ba831f3e3ba9295e368521a87/lightstep/tracer.py#L76-L77) the changes in `tracer.py`. I also [updated](https://github.com/lightstep/lightstep-tracer-python/blob/70da71a1eb75b32ba831f3e3ba9295e368521a87/examples/http/context_in_headers.py#L151-L154) the example to explicitly call `register_propagator` twice to register the B3 propagator for `TEXT_MAP` and `HTTP_HEADERS` respectively.

Just to be sure, @carlosalberto, you mentioned this in your previous comment:
```
# replaces the default propagator with the B3 one
tracer.register_propagator(b3_propagator)
```
But is not possible to call `register_propagator` with only one argument, the format name and propagator instance are needed.

Please let me know if this is the right implementation.

Thanks!

Reply:
> But is not possible to call register_propagator with only one argument, the format name and propagator instance are needed.

Yeah, this is fine ;)